### ISSUE TITLE
Deleted external-secrets still showing metrics with status ready

### ISSUE THREAD

**Issue by netic-jsj** [external-secrets/external-secrets#471]
Im trying to setup monitoring of external secrets. I can se that in issue #231, the metrics has added the stage 'Deleted'.
Deleting a externalsecret creates a new metric with condition Deleted but the old metric with condition Ready is still there. Is this as Intended?

This is testet with version 0.3.5 and 0.3.7.

Initial metrics
```
externalsecret_status_condition{condition="Ready",name="cortex-s3-auth",namespace="xxxxx-cortex-system",status="True"} 1
externalsecret_status_condition{condition="Ready",name="oauth2-proxy",namespace="xxxxx-ingress-system",status="True"} 1
externalsecret_status_condition{condition="Ready",name="tidydns-auth",namespace="xxxxx-security-system",status="True"} 1
```

Created 3 externalsecrets
```
externalsecret_status_condition{condition="Ready",name="cortex-s3-auth",namespace="xxxxx-cortex-system",status="True"} 1
externalsecret_status_condition{condition="Ready",name="oauth2-proxy",namespace="xxxxx-ingress-system",status="True"} 1
externalsecret_status_condition{condition="Ready",name="secret1",namespace="default",status="False"} 1
externalsecret_status_condition{condition="Ready",name="secret2",namespace="default",status="False"} 1
externalsecret_status_condition{condition="Ready",name="secret3",namespace="default",status="False"} 1
externalsecret_status_condition{condition="Ready",name="tidydns-auth",namespace="xxxxx-security-system",status="True"} 1
```

Deleted 3 externalsecrets
```
externalsecret_status_condition{condition="Deleted",name="secret1",namespace="default",status="False"} 1
externalsecret_status_condition{condition="Deleted",name="secret2",namespace="default",status="False"} 1
externalsecret_status_condition{condition="Deleted",name="secret3",namespace="default",status="False"} 1
externalsecret_status_condition{condition="Ready",name="cortex-s3-auth",namespace="xxxxx-cortex-system",status="True"} 1
externalsecret_status_condition{condition="Ready",name="oauth2-proxy",namespace="xxxxx-ingress-system",status="True"} 1
externalsecret_status_condition{condition="Ready",name="secret1",namespace="default",status="False"} 1
externalsecret_status_condition{condition="Ready",name="secret2",namespace="default",status="False"} 1
externalsecret_status_condition{condition="Ready",name="secret3",namespace="default",status="False"} 1
externalsecret_status_condition{condition="Ready",name="tidydns-auth",namespace="xxxxx-security-system",status="True"} 1
```

Pod restarted
```
externalsecret_status_condition{condition="Ready",name="cortex-s3-auth",namespace="xxxxx-cortex-system",status="True"} 1
externalsecret_status_condition{condition="Ready",name="oauth2-proxy",namespace="xxxxx-ingress-system",status="True"} 1
externalsecret_status_condition{condition="Ready",name="tidydns-auth",namespace="xxxxx-security-system",status="True"} 1
```

**Comment by hjorth** [comment]
I was about to do exactly the same after upgrading from 0.2.2 to 3.5.7.

Just to add to this, it looks, like if you delete a secret and recreate it - it will also still be visible as deleted.

**Comment by dklbe** [comment]
Having also issues with the ExternalSecrets custom resource after deletion/re-creation not showing the right status. I'd really appreciate a fix for that.

**Comment by moolen** [comment]
This got fixed in #612
